### PR TITLE
Increase the `z-index` of the jump targets

### LINKED ISSUE #6746: Jump-targets have higher z-index than dropdown-menu
### Affected version(s)

5.3.0-RC1

### Description

<h3>Bug</h3>

- The z-index of the jump-targets (3) is higher than the user-menu (2), hence overlapping

image

This does not occur sub 991px due to `#main` using `will-change: transform` [(See Stacking-context)](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_positioned_layout/Understanding_z-index/Stacking_context)

Having a `z-index: 1` on the jump-targets should be sufficient.

### PR DESCRIPTION
**Pull request by zoglo**
Fixes #6746 

A z-index of 1 should be sufficient for `.jump-targets`, we could also raise the z-index of the dropdown-menu instead.

### CODE REVIEW

**Comment by leofeyer** [comment]
I doubt that this works. Do the jump targets still overlay the rich text editor with `z-index: 1`?

**Comment by zoglo** [comment]
> I doubt that this works. Do the jump targets still overlay the rich text editor with `z-index: 1`?

Fixed in [3a72ff](https://github.com/contao/contao/pull/6747/commits/3a72ff336438216243211e6939e8b79de39db0cd)

See:
image
image


@leofeyer Seems like the codecov checks failed due to a 503 🤔? Could you re-run it from your side?
```
[2024-01-19T17:21:57.096Z] ['error'] There was an error running the uploader: Error uploading to [https://codecov.io:](https://codecov.io/) Error: There was an error fetching the storage URL during POST: 503 - upstream connect error or disconnect/reset before headers. reset reason: connection failure```

File: core-bundle/contao/themes/flexible/styles/main.css [hunk 1]
```diff
@@ -169,7 +169,7 @@ body.popup {
 	background: var(--content-bg);
 	border: 1px solid var(--content-border);
 	box-shadow: 0 1px 6px rgba(0,0,0,.2);
-	z-index: 2;
+	z-index: 4;
```
Comment (ausi) [inline core-bundle/contao/themes/flexible/styles/main.css]:
```suggestion
	z-index: 4; /* above .jump-targets */
```

Just a nitpick: I think it’s good practise to document all `z-index` values greater than `1`.

Comment (zoglo) [reply core-bundle/contao/themes/flexible/styles/main.css]: Taking https://github.com/contao/contao/pull/6747#issuecomment-1900771908 into consideration, I checked the other ones and left comments for higher z-index as well.

The comments were added in https://github.com/contao/contao/pull/6747/commits/738329c76b8b17edd97b7d1e3670bf039c05aa28 @ausi

**Comment by zoglo** [comment]
Added more comments for higher z-index as well

**Review by ausi — DISMISSED** [review]

File: core-bundle/contao/themes/flexible/styles/main.css [hunk 2]
```diff
@@ -742,7 +742,7 @@ h2.sub_headline {
 	border-bottom: 1px solid var(--content-border);
 	position: sticky;
 	top: 0;
-	z-index: 3;
+	z-index: 3; /* Above TinyMCE */
```
Comment (leofeyer) [inline core-bundle/contao/themes/flexible/styles/main.css:745]: There is also an _Above TinyMCE_ comment in line 853, however, it only uses `z-index: 2` and not `3`.  So either the value is wrong here or the comment is incomplete.

Comment (zoglo) [reply core-bundle/contao/themes/flexible/styles/main.css:745]: I was simply referring to https://github.com/contao/contao/pull/6747#discussion_r1460344690 and commented every `z-index` in the file.

Comment (leofeyer) [reply core-bundle/contao/themes/flexible/styles/main.css:745]: Yes, but why do we use `z-index: 3` here and only `z-index: 2` in line 853? According to the comment, both do the same thing.

**Review by leofeyer — COMMENTED** [review]
The changes work, however, the TinyMCE popup menus use `z-index: 1300`:



---

Do we care about that? We could fix it by using `z-index: 1332` for the drop-down menu and `z-index: 1331` for the jump to links and the submit buttons:

**Comment by ausi** [comment]
> Do we care about that?

I would leave it as it is, as these menus only open on direct user interaction. For small screens this would even increase the usability of the menus.

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @zoglo.

**Pull request merged**